azurelinuxagent/ga/monitor.py

```diff
@@ -178,8 +179,8 @@ def init_sysinfo(self):
         try:
             ram = self.osutil.get_total_mem()
             processors = self.osutil.get_processor_cores()
-            self.sysinfo.append(TelemetryEventParam("RAM", ram))
-            self.sysinfo.append(TelemetryEventParam("Processors", processors))
+            self.sysinfo.append(TelemetryEventParam("RAM", int(ram)))
```

Comment:
shouldn't the conversion be done by get_total_mem?

Reply:
The casting is done for the purposes of solving current and potential schema issues. I am keeping the original type of `get_total_mem` in case we have any other uses which require its origin type (float/int, depending on the platform).

Reply:
OK. Shouldn't then those casts also be done in add_default_parameters_to_event, _add_event, etc.

Reply:
Yes. Outside the purview of this PR and I'll validate if it requires any changes and send another PR for that if needed.

Reply:
Did the change for PerfCounters and Add_Event. Will make the change for Add_log_event in the send logs PR.